stage0/src/sev.rs

```diff
@@ -96,18 +84,44 @@ pub fn init_ghcb(
 
 pub fn deinit_ghcb(snp: bool, encrypted: u64) {
     let ghcb_addr = GHCB_WRAPPER.get().unwrap().lock().get_gpa().as_u64();
+    unshare_page(PhysAddr::new(ghcb_addr), snp, encrypted);
+}
 
+/// Shares a single 4KiB page containing the address with the hypervisor.
+pub fn share_page(address: PhysAddr, snp: bool, encrypted: u64) {
```

Comment:
These functions should probably take a `Page<Size4KiB>` as an argument instead of a raw PhysAddr, but I'm fine delegating that clean-up to some later point in time.

Reply:
Good point. Done, at least an initial pass at making it a bit better.